src/views/PassportView.vue

```diff
@@ -143,13 +145,15 @@ export default {
       if(result && result != null){
         this.data = result;
       }else{
-        this.loading = false;
         this.error = true;
-        this.errorObj.title = "Timeout! Failed to return passport!";
-        this.errorObj.description = "We are sorry, it took too long to retrieve the passport.";
+        if(this.errorObj.title == null){
+          this.errorObj.title = "Timeout! Failed to return passport!";

```

Comment:
The error handling was always giving timeout, and all the exceptions were not handled correctly